server/public/model/config.go

```diff
@@ -3034,14 +3034,13 @@ type CloudSettings struct {
 }
 
 func (s *CloudSettings) SetDefaults() {
-	if s.CWSURL == nil {
-		switch GetServiceEnvironment() {
-		case ServiceEnvironmentProduction:
-			s.CWSURL = NewString(CloudSettingsDefaultCwsURL)
-		case ServiceEnvironmentTest, ServiceEnvironmentDev:
-			s.CWSURL = NewString(CloudSettingsDefaultCwsURLTest)
-		}
+	switch GetServiceEnvironment() {
+	case ServiceEnvironmentProduction:
+		s.CWSURL = NewString(CloudSettingsDefaultCwsURL)
+	case ServiceEnvironmentTest, ServiceEnvironmentDev:
+		s.CWSURL = NewString(CloudSettingsDefaultCwsURLTest)
 	}
+
```

Comment:
Would we consider "removing" the configuration altogether and doing this check closer to the code using it?

Reply:
Definitely something to consider long term. Some things are in flux around Self Serve in both Cloud and Self Hosted which depending on the outcome would have rippling effects on areas of code just like your suggestion. Until those decisions shake out I'm holding off on any more significant changes